system/jlib/jfile.cpp

```diff
@@ -5633,22 +5633,23 @@ IFileIO *createUniqueFile(const char *dir, const char *prefix, const char *ext,
     CDateTime dt;
     dt.setNow();
     unsigned t = (unsigned)dt.getSimple();
-    if (dir)
-    {
-        filename.append(dir);
-        addPathSepChar(filename);
-    }
-    if (prefix && *prefix)
-        filename.append(prefix);
-    else
-        filename.append("uniq");
+    unsigned attempts = 5; // max attempts
     if (!ext || !*ext)
         ext = "tmp";
-    filename.appendf("_%" I64F "x.%x.%x.%s", (__int64)GetCurrentThreadId(), (unsigned)GetCurrentProcessId(), t, ext);
-    OwnedIFile iFile = createIFile(filename.str());
-    unsigned attempts = 5; // max attempts
     for (;;)
     {
+        filename.clear();
+        if (dir)
+        {
+            filename.append(dir);
+            addPathSepChar(filename);
+        }
+        if (prefix && *prefix)
+            filename.append(prefix);
+        else
+            filename.append("uniq");
```

Comment:
since the path and prefix are constant, I might be tempted to keep filename prep. before the loop,
but record it's length and reset to that pos on a retry [ filename.setLength(tailPos); ]

Reply:
I was going for simplicity rather than efficiency - this isn't expected to clash, and it isn't time critical.

Reply:
yep, agreed